src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,177 @@
+                                          where p.IsIndexer
+                                          select SymbolCompletionItem.CreateWithSymbolId(
+                                              displayText: $"[{string.Join(", ", p.Parameters.Select(p => p.Type.ToMinimalDisplayString(semanticModel, position)))}]", // The type to convert to
+                                              symbols: ImmutableList.Create(p),
+                                              rules: CompletionItemRules.Default,
+                                              contextPosition: position,
+                                              properties: CreateCompletionHandlerProperty(CompletionHandlerIndexer));
+                            context.AddItems(allExplicitConversions.Union(indexer));
+                        }
+                    }
+                }
+            }
+        }
+
+        private static ExpressionSyntax? GetExpressionOfInvocation(SyntaxToken dotToken)
+        {
+            Debug.Assert(dotToken.IsKind(SyntaxKind.DotToken));
+            return dotToken.Parent switch
+            {
+                MemberAccessExpressionSyntax memberAccess => memberAccess.Expression,
+                MemberBindingExpressionSyntax { Parent: ConditionalAccessExpressionSyntax conditionalAccess } _ => conditionalAccess.Expression,
+                _ => null,
+            };
+        }
+
+        internal override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, TextSpan completionListSpan, char? commitKey, bool disallowAddingImports, CancellationToken cancellationToken)
+        {
+            if (item.Properties.TryGetValue(CompletionHandlerPropertyname, out var value))
+            {
+                var completionChange = value switch
+                {
+                    CompletionHandlerConversion => await HandleConversionChangeAsync(document, item, cancellationToken).ConfigureAwait(false),
+                    CompletionHandlerIndexer => await HandleIndexerChangeAsync(document, item, cancellationToken).ConfigureAwait(false),
+                    _ => null,
+                };
+                if (completionChange is { })
+                {
+                    return completionChange;
+                }
+            }
+
+            return await base.GetChangeAsync(document, item, completionListSpan, commitKey, disallowAddingImports, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<CompletionChange?> HandleConversionChangeAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var symbols = await SymbolCompletionItem.GetSymbolsAsync(item, document, cancellationToken).ConfigureAwait(false);
+            var position = SymbolCompletionItem.GetContextPosition(item);
+            var symbol = symbols.Single() as IMethodSymbol;
+            var convertToType = symbol!.ReturnType;
```

Comment:
don't use `as` and then suppress away null warnings.  either you're using `as` because you expect it may fail (and should then check for that), or you expect it to never fail, in which case don't use `as` :)